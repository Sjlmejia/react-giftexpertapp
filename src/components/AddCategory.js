import React, { useState } from "react";
import PropTypes from 'prop-types';
export const AddCategory = ({setCategories}) => {

  let [inputValue, setInputValue] = useState('')
  const onChangeValue = (value) => {
    setInputValue(value.target.value)
    console.log('llamado')
  }
  const handleSubmit = (e) => {
    e.preventDefault();
    if( inputValue.trim().length > 2) {
      setCategories(cats => [inputValue, ...cats]);
      setInputValue('');
    }
  }
  return (
    <form onSubmit= {handleSubmit}>
      <input
        type= "text"
        value= {inputValue}
        onChange= {(e)=>{onChangeValue(e)}}
      ></input>
    </form>
  )
}

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}