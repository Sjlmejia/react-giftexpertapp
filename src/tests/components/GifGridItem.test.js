import React from 'react';
import {shallow} from 'enzyme'
import { GifGridItem } from '../../components/GifGridItem';

describe('Pruebas en <GiftGridItem />', () => {
  const title = 'titulo';
  const url = 'es un url'
  const wrapper = shallow(<GifGridItem title={title} url={url}  />)

  test('debe mostrar el componente', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('debe tener un parrado con el titulo', () => {
    const p = wrapper.find('p');
    expect( p.text().trim()).toBe( title)
  });

  test('debe tener imagen igual al url y alt', () => {
    
    const img = wrapper.find('img');
    expect(img.prop('src')).toBe(url)
    expect(img.prop('alt')).toBe(title)
  });

  test('debe tener la clase', () => {
    const div = wrapper.find('div');
    const clasName = div.prop('className');

    expect(clasName.includes('card')).toBe(true)
  });
});