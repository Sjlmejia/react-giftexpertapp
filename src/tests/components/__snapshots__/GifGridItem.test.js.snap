// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftGridItem /> debe mostrar el componente 1`] = `
<div
  className="card animate__animated animate__bounce"
>
  <img
    alt="titulo"
    src="es un url"
  />
  <p>
    titulo
  </p>
</div>
`;
